drivers/gpio/Kconfig.intel

```diff
@@ -21,3 +21,26 @@ config GPIO_INTEL_CHECK_PERMS
 	  the driver allows manipulating the pin.
 
 	  Say y if unsure.
+
+config GPIO_INTEL_INT_EN_OFFSET
+	hex "Offset to Interrupt Enable register from Interrupt Status register"
+	default 0x20
+	depends on GPIO_INTEL && \
+		   $(dt_compat_any_has_prop,$(DT_COMPAT_INTEL_GPIO),acpi-hid)
+	help
+	  This option enables to add proper offset to access
+	  GPIO_INT_EN register from GPIO_INT_STAT register,
+	  as it is observed that the offset varies from
+	  platform to platform.
+
+config GPIO_INTEL_INT_STAT_OFFSET
```

Comment:
I wonder if these two actually belong to devicetree, as they are per platform (so they would be added to `dts/bindings/gpio/intel,gpio.yaml`).

Reply:
Typically, we would describe that in the devicetree instance, but there's no good way of setting the same value across a number of devicetree nodes. Only thing I can think of is having a parent node with this property, but that would be kind of superficial.

Is this GPIO controller supported in Linux? if yes, how does the Linux kernel handle/provide this offset?

Reply:
This GPIO controller is supported in Linux, a different ACPI method is used to fetch this offset directly instead of deriving from another offset like what we are doing here. That ACPI method is not enabled in zephyr. [Linux Ref](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/pinctrl/intel/pinctrl-intel-platform.c?h=v6.17#n114)